x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -290,6 +292,9 @@ export const postActionsConnectorExecuteRoute = (
         } catch (err) {
           logger.error(err);
           const error = transformError(err);
+          if (onLlmResponse) {
+            onLlmResponse(error.message, {}, true);
+          }
```

Comment:
Bug fixed! 🙌 

Note that with streaming enabled, there is a slight flicker from the response error message initially being rendered without `isError`:

<p align="center">
  <img width="700" src="https://github.com/elastic/kibana/assets/2946766/f5d5894e-83c3-4012-8227-93dadf3516b0" />
</p> 

Looks like the streaming response still returns `200` OK, and so message first gets rendered from response w/o `isError`, then the persistence version with it immediately gets pushed down to sync the conversation:


<p align="center">
  <img width="700" src="https://github.com/elastic/kibana/assets/2946766/c980b9f3-b90f-4df2-8e32-478bb1689aac" />
</p> 


Is it possible to get the streaming request to `/_execute` to return a status other than `200` here? Otherwise we'd probably need to parse the `payload` for errors to fix (since I don't think we can modify the final payload, right?).

Reply:
Discussed over slack. Non-LangChain streaming error responses will be 500. LangChain streaming error responses currently 200. This is a known issue. If we allow the `openai-functions` `exectuor.invoke` to error, Kibana crashes. Will look into that further in another issue